### PR TITLE
Configure additional swift services

### PR DESCRIPTION
**Pull request by vuntz**
There's one tempest test at least depending on the container sync.

### CODE REVIEW

**Comment by vuntz** [comment]
Gating is failing due to a packaging bug in swift; I fixed that but it will take some time to reach gating...

File: chef/cookbooks/swift/recipes/storage.rb [hunk 1]
```diff
@@ -44,7 +44,11 @@
 
 storage_ip = Swift::Evaluator.get_ip_by_type(node,:storage_ip_expr)
 
-%w{account-server object-server container-server}.each do |service|
+env_filter = " AND swift_config_environment:#{node[:swift][:config][:environment]}"
+compute_nodes = search(:node, "roles:swift-ring-compute#{env_filter}")
```
Comment (aplanas) [inline chef/cookbooks/swift/recipes/storage.rb]: Style / readability question, is not more clear this?

```ruby
env_filter = "swift_config_environment:#{node[:swift][:config][:environment]}"
compute_nodes = search(:node, ["roles:swift-ring-compute", env_filter].join(" AND "))
proxy_nodes = search(:node, ["roles:swift-proxy", env_filter].join(" AND "))
```

Comment (vuntz) [inline chef/cookbooks/swift/recipes/storage.rb]: You are correct; I actually think we have a helper that we can use for this and that makes things look simpler.

File: chef/data_bags/crowbar/template-swift.schema [hunk 2]
```diff
@@ -118,6 +118,10 @@
               "generate_certs": { "type" : "bool", "required" : true },
               "insecure": { "type" : "bool", "required" : true }
             }},
+            "container_sync": { "type": "map", "required": true, "mapping": {
+              "key": { "type" : "str", "required" : true },
+              "key2": { "type" : "str", "required" : true }
```
Comment (aplanas) [inline chef/data_bags/crowbar/template-swift.schema]: do we need migrations for `key` and `key2`?

Comment (vuntz) [inline chef/data_bags/crowbar/template-swift.schema]: Gah, I forgot to git add that. Fixed!

File: chef/cookbooks/swift/recipes/storage.rb (hunk 1 above)
Comment (vuntz) [inline chef/cookbooks/swift/recipes/storage.rb]: fixed

**Comment by aplanas** [comment]
LGTM +1

**Comment by aplanas** [comment]
+1

**Comment by jsuchome** [comment]
Not tested, but looks good to me. +1

**Pull request merged**